packages/block-library/src/columns/editor.scss

```diff
@@ -89,29 +93,49 @@
 
 			// Beyond mobile, allow 2 columns.
 			@include break-small() {
-				flex-basis: 50%;
+				// Size should be half the available width, minus the margins added below.
+				// In the editor, we also need to account for additional block padding.
+				flex-basis: calc(50% - #{ $grid-size-large } - #{$block-padding * 2});
 				flex-grow: 0;
 			}
 
 			// Add space between columns. Themes can customize this if they wish to work differently.
 			// This has to match the same padding applied in style.scss.
 			// Only apply this beyond the mobile breakpoint, as there's only a single column on mobile.
+
+			// 2 columns at this breakpoint, and items beyond the second will wrap.
 			@include break-small() {
 				&:nth-child(odd) {
-					margin-right: $grid-size-large * 2;
+					margin-right: $grid-size-large + $block-padding;
 				}
+
 				&:nth-child(even) {
-					margin-left: $grid-size-large * 2;
+					margin-left: $grid-size-large + $block-padding;
 				}
 			}
 
-			@include break-small() {
+			// Beyond the medium breakpoint, show the user-set amount of columns.
+			@include break-medium() {
+				// Because columns do not wrap, reset the column margin.
+				margin-bottom: initial;
```

Comment:
Looks like postcss or something else is actually converting it to `0` in the final CSS anyway. Still probably a good idea to avoid the abstraction here.

Reply:
> Still probably a good idea to avoid the abstraction here.

Makes sense to me. I guess using `initial` doesn't really _get us_ anything.